apps/buildog/src/app/(blog-pages)/blog/page.tsx

```diff
@@ -1,85 +1,167 @@
 type Blog = {
   cards: {
     id: string;
     value: string;
     lock: boolean;
   }[];
+  title: string;
   filename: string;
   text: string;
+  status: string;
+  tags: string[];
+  slug?: string;
 };
 
-type Blogs = {
-  [filename: string]: Blog;
-};
+// Define the Blogs type as an array of Blog objects
+type Blogs = Blog[];
+
+// Sample data
+const dummyBlogs: Blogs = [
+  {
+    cards: [
+      { id: "card1", value: "Card 1 content", lock: false },
+      { id: "card2", value: "Card 2 content", lock: true },
+    ],
+    title: "My first blog",
+    filename: "blog1.md",
+    text: "This is the text content for blog 1. It provides details about the blog post, discussing its main topics and giving an overview of what the user can expect when reading the full article.",
+    status: "Published",
+    tags: ["React", "JavaScript", "Web Development"],
+  },
+  {
+    cards: [
+      { id: "card3", value: "Card 3 content", lock: false },
+      { id: "card4", value: "Card 4 content", lock: true },
+    ],
+    title: "My third blog",
+    filename: "blog2.md",
+    text: "This article delves into the core aspects of web development with React, exploring the framework's features, best practices, and common pitfalls. It offers a comprehensive guide to mastering React, complete with examples and tips to enhance your coding skills and build robust applications.",
+    status: "Draft",
+    tags: ["Draft", "Planning", "Concept"],
+  },
+  {
+    cards: [
+      { id: "card5", value: "Card 5 content", lock: false },
+      { id: "card6", value: "Card 6 content", lock: true },
+    ],
+    title: "My second blog",
+    filename: "blog3.md",
+    text: "In this post, we examine the fundamentals of JavaScript and its role in modern web development. From basic syntax to advanced techniques, this guide covers essential concepts and provides practical advice for writing clean, efficient code and leveraging JavaScript's full potential.",
+    status: "Scheduled",
+    tags: ["Archiving", "Content Management", "Best Practices"],
+  },
+];
 
 export default function Page() {
-  const [blogs, setBlogs] = useState<Blogs>();
+  const [blogs, setBlogs] = useState<Blogs>(dummyBlogs);
   const [open, setOpen] = useState(false);
 
-  useEffect(() => {
-    if (typeof window !== "undefined") {
-      const localStorageBlogs = JSON.parse(localStorage.getItem("blogs") as string);
-      setBlogs(localStorageBlogs);
-    }
-  }, []);
-
-  const deleteBlog = (filenameToDelete: string) => {
+  const deleteBlog = (blogToDelete: string) => {
     setOpen(!open);
+
     setBlogs((prevBlogs) => {
-      const updatedBlogs = { ...prevBlogs };
-      delete updatedBlogs[filenameToDelete];
-      if (typeof window !== "undefined") {
-        localStorage.setItem("blogs", JSON.stringify(updatedBlogs));
-      }
+      if (!prevBlogs) return prevBlogs;
+
+      // Filter out the blog to delete
+      const updatedBlogs = prevBlogs.filter((blog) => blog.title !== blogToDelete);
```

Comment:
Filter may not a good idea here. We will make a backend endpoint for deleting. Right now no need a change.

Reply:
Looking title to delete is not a best idea. We have to generate unique id to delete blog.